### PR TITLE
Implement ocrtypes.Database interface

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/internal/cltest/factories.go [hunk 1]
```diff
@@ -787,3 +787,9 @@ func MustInsertRandomKey(t *testing.T, store *strpkg.Store) models.Key {
 	require.NoError(t, store.CreateKeyIfNotExists(k))
 	return k
 }
+
+func MustInsertOffchainreportingOracleSpec(t *testing.T, store *strpkg.Store) models.OffchainreportingOracleSpec {
+	spec := models.OffchainreportingOracleSpec{}
```
Comment (j16r) [inline core/internal/cltest/factories.go:792]: Might want a `t.Helper()` here.

File: core/store/offchainreportingdb/database_test.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+		1,
+		2,
+	}
+
+	t.Run("stores and retrieves pending transmissions", func(t *testing.T) {
+		p := ocrtypes.PendingTransmission{
+			time.Now(),
+			ocrtypes.Observation(big.NewInt(41)),
+			[]byte{0, 2, 3},
+			[][32]byte{[32]byte{0, 0, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 4, 5, 6, 7, 8, 9, 0, 1, 2}},
```
Comment (j16r) [inline core/store/offchainreportingdb/database_test.go:173]: Could we put these in hex and convert them to bytes maybe with a test helper?

File: core/internal/cltest/factories.go (hunk 1 above)
Comment (samsondav) [reply core/internal/cltest/factories.go:792]: What does it do?

Comment (j16r) [reply core/internal/cltest/factories.go:792]: If `require` fails, it'll print the error in your test, rather than on line 793.

**Review by j16r — APPROVED** [review]

**Review by j16r — APPROVED** [review]

**Pull request merged**